nav2_costmap_2d/src/costmap_subscriber.cpp

```diff
@@ -16,88 +16,210 @@
 #include <memory>
 
 #include "nav2_costmap_2d/costmap_subscriber.hpp"
+#include "nav2_costmap_2d/cost_values.hpp"
+
+using nav2_costmap_2d::NO_INFORMATION;
+using nav2_costmap_2d::LETHAL_OBSTACLE;
+using nav2_costmap_2d::FREE_SPACE;
 
 namespace nav2_costmap_2d
 {
 
 CostmapSubscriber::CostmapSubscriber(
   nav2_util::LifecycleNode::SharedPtr node,
-  const std::string & topic_name)
+  const std::string & topic_name,
+  bool use_raw)
 : CostmapSubscriber(node->get_node_base_interface(),
     node->get_node_topics_interface(),
     node->get_node_logging_interface(),
-    topic_name)
+    node->get_node_parameters_interface(),
+    topic_name,
+    use_raw)
 {}
 
 CostmapSubscriber::CostmapSubscriber(
   rclcpp::Node::SharedPtr node,
-  const std::string & topic_name)
+  const std::string & topic_name,
+  bool use_raw)
 : CostmapSubscriber(node->get_node_base_interface(),
     node->get_node_topics_interface(),
     node->get_node_logging_interface(),
-    topic_name)
+    node->get_node_parameters_interface(),
+    topic_name,
+    use_raw)
 {}
 
 CostmapSubscriber::CostmapSubscriber(
   const rclcpp::node_interfaces::NodeBaseInterface::SharedPtr node_base,
   const rclcpp::node_interfaces::NodeTopicsInterface::SharedPtr node_topics,
   const rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr node_logging,
-  const std::string & topic_name)
+  const rclcpp::node_interfaces::NodeParametersInterface::SharedPtr node_parameters,
+  const std::string & topic_name,
+  bool use_raw)
 : node_base_(node_base),
   node_topics_(node_topics),
   node_logging_(node_logging),
-  topic_name_(topic_name)
+  node_parameters_(node_parameters),
+  topic_name_(topic_name),
+  use_raw_(use_raw)
 {
-  costmap_sub_ = rclcpp::create_subscription<nav2_msgs::msg::Costmap>(node_topics_, topic_name_,
+  if (use_raw_) {
+    costmap_raw_sub_ = rclcpp::create_subscription<nav2_msgs::msg::Costmap>(node_topics_,
+        topic_name_,
+        rclcpp::QoS(rclcpp::KeepLast(1)).transient_local().reliable(),
```

Comment:
Not blocking: but we talked above about the QoS settings. I agree they line up with the publisher, but I'm more asking "is this the right QoS we _should_ be using?" and if the answer is no, can you file a ticket with your proposal. I can do either way on this.